packages/webapp/src/components/Task/AddSoilAmendmentProducts/ProductCard/index.tsx

```diff
@@ -87,11 +93,47 @@ const SoilAmendmentProductCard = ({
   return (
     <div className={styles.productCard}>
       {onRemove && <SmallButton onClick={onRemove} className={styles.removeButton} />}
+      <Controller
+        control={control}
+        name={PURPOSES}
+        rules={{ required: true }}
+        render={({ field: { onChange, value: selectedOptions = [] } }) => (
+          <ReactSelect
+            isMulti
+            value={purposeOptions.filter(({ value }) => selectedOptions.includes(value))}
+            isDisabled={isReadOnly}
+            label={t('ADD_TASK.SOIL_AMENDMENT_VIEW.PURPOSE')}
+            options={purposeOptions}
+            onChange={(e) => {
+              onChange(e);
+              const newPurposes = e.map(({ value }) => value);
+              setValue(PURPOSES, newPurposes, { shouldValidate: true });
+            }}
+          />
+        )}
+      />
+      {purposes?.includes(otherPurposeId) && (
+        <>
+          {/* @ts-ignore */}
+          <Input
+            label={t('ADD_TASK.SOIL_AMENDMENT_VIEW.OTHER_PURPOSE')}
+            name={PRODUCT_FIELD_NAMES.OTHER_PURPOSE}
+            disabled={isReadOnly}
+            hookFormRegister={register(PRODUCT_FIELD_NAMES.OTHER_PURPOSE)}
+            optional
+          />
+        </>
+      )}
       <div>
         <Controller
           control={control}
           name={PRODUCT_ID}
-          rules={{ required: true }}
+          rules={{
+            required: true,
+            validate: (value) => {
```

Comment:
Using this + the form `isValid` state is indeed quite improved (simplified) from `getInvalidProductsUpdater` 🙂  I like it!